### PR TITLE
Collect ES stack traces and report all health check errors

### PR DESCRIPTION
**Pull request by pebrc**
During the investigation of https://github.com/elastic/cloud-on-k8s/issues/5865 I saw a few of these 4xx errors that would like to better understand: 

```
[23][null][BAD_REQUEST][application/json;charset=utf-8][351] sent response to [Netty4HttpChannel{localAddress=/10.244.3.206:9200, remoteAddress=/10.244.2.3:46366}] success [true]
```
These are health check responses to our e2e test Pod. I would like to better understand what is going on there. Also `error_trace` is a good idea in general for tests and dev mode.

### CODE REVIEW

File: pkg/controller/elasticsearch/client/base.go [hunk 1]
```diff
@@ -129,7 +131,18 @@ func (c *baseClient) request(
...
+		if err != nil {
+			return err
+		}
+		q := url.Query()
+		q.Set("error_trace", "true")
+		url.RawQuery = q.Encode()
+		rawURL = url.String()
+	}
+	request, err := http.NewRequest(method, rawURL, body) //nolint:noctx
 	if err != nil {
```
Comment (naemono) [inline pkg/controller/elasticsearch/client/base.go:146]: Why don't you wait to check to see if debug is enabled until after you generate the request? Then you won't have to parse the URL and have that additional error logic.  Something like

```
q := request.URL.Query()
q.Add("error_trace", "true")
request.URL.RawQuery = q.Encode()
```

File: test/e2e/test/elasticsearch/steps_mutation_test.go [hunk 2]
```diff
@@ -22,13 +23,13 @@ func Test_clusterUnavailability(t *testing.T) {
 	require.False(t, cu.hasExceededThreshold())
 
 	// mark the cluster as available, we're still below the threshold
-	cu.markUnavailable()
+	cu.markUnavailable(errors.New("Connection refused"))
```
Comment (thbkrkr) [inline test/e2e/test/elasticsearch/steps_mutation_test.go:26]: It doesn't matter here but error strings should not be capitalized (unless they start with proper names or acronyms). To keep the habit?

```suggestion
	cu.markUnavailable(errors.New("connection refused"))
```

File: test/e2e/test/elasticsearch/steps_mutation_test.go [hunk 3]
```diff
@@ -22,13 +23,13 @@ func Test_clusterUnavailability(t *testing.T) {
...
 	require.False(t, cu.start.IsZero())
 	require.False(t, cu.hasExceededThreshold())
 
 	// marking as unavailable again should not change the start time
 	initialStartTime := cu.start
-	cu.markUnavailable()
+	cu.markUnavailable(errors.New("Context deadline exceeded"))
```
Comment (thbkrkr) [inline test/e2e/test/elasticsearch/steps_mutation_test.go:32]:
```suggestion
	cu.markUnavailable(errors.New("context deadline exceeded"))
```

File: test/e2e/test/elasticsearch/steps_mutation_test.go [hunk 4]
```diff
@@ -38,7 +39,7 @@ func Test_clusterUnavailability(t *testing.T) {
 	require.False(t, cu.hasExceededThreshold())
 
 	// simulate a lower threshold we should have exceeded
-	cu.markUnavailable()
+	cu.markUnavailable(errors.New("Something else"))
```
Comment (thbkrkr) [inline test/e2e/test/elasticsearch/steps_mutation_test.go:42]:
```suggestion
	cu.markUnavailable(errors.New("something else"))
```

**Review by thbkrkr — APPROVED** [review]
LGTM

**Review by naemono — APPROVED** [review]
Was waiting for checks to pass but 👍

**Pull request merged**